### PR TITLE
what-is-rccl

### PR DESCRIPTION
**Pull request by randyh62**
## Details
___Do not mention proprietary info or link to internal work items in this PR.___

**Work item:** _"Internal", or link to GitHub issue (if applicable)._

**What were the changes?**  
_One sentence describing the work done._

**Why were the changes made?**  
_Explain the motivation behind the work. Provide any publicly-available historical context._

**How was the outcome achieved?**  
_Technical details behind the work. Explain any publicly-available hardware peculiarities._

**Additional Documentation:**  
_What else should the reviewer know?_

## Approval Checklist
___Do not approve until these items are satisfied.___
- [ ] Verify the CHANGELOG has been updated, if
  - there are any NCCL API version changes,
  - any changes impact library users, and/or
  - any changes impact any other ROCm library.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -4,24 +4,28 @@
...
-The provided collective communication routines are implemented using Ring and Tree algorithms. They are optimized to achieve high bandwidth and low latency by leveraging topology awareness, high-speed interconnects, and RDMA based collectives. 
+The ROCm Communication Collectives Library (RCCL) is a stand-alone library that provides multi-GPU and multi-node collective communication primitives optimized for AMD GPUs.
+It implements routines such as ``all-reduce``, ``all-gather``, ``reduce``, ``broadcast``, ``reduce-scatter``, ``gather``, ``scatter``, ``all-to-allv``, and ``all-to-all`` as well as direct point-to-point (GPU-to-GPU) send and receive operations. There is also initial support for direct GPU-to-GPU send and receive operations.  It has been optimized to achieve high bandwidth on platforms using PCIe, xGMI as well as networking using InfiniBand Verbs or TCP/IP sockets. RCCL supports an arbitrary number of GPUs installed in a single node or multiple nodes, and can be used in either single- or multi-process (e.g., MPI) applications.
```
Comment (nileshnegi) [inline docs/index.rst]: is this a repetition?

File: docs/index.rst [hunk 2]
```diff
@@ -4,24 +4,28 @@
...
 
-RCCL utilizes PCIe and xGMI high-speed interconnects for intra-node communication as well as InfiniBand, RoCE, and TCP/IP for inter-node communication.
-It supports an arbitrary number of GPUs installed in a single-node or multi-node platform and can be easily integrated into single- or multi-process (e.g., MPI) applications.
+The collective operations are implemented using ring and tree algorithms, and have been optimized for throughput and latency by leveraging topology awareness, high-speed interconnects, and RDMA based collectives. For best performance, small operations can be either batched into larger operations or aggregated through the API.
```
Comment (nileshnegi) [inline docs/index.rst]: can we make this `Ring` and `Tree`, please?

**Review by corey-derochie-amd — COMMENTED** [review]
We need to ensure that these docs are kept in sync with the README, and are an exact match. Is there something else we should do here?

File: docs/index.rst (hunk 1 above)
Comment (randyh62) [reply docs/index.rst]: Removed the redundancy

**Comment by randyh62** [comment]
> We need to ensure that these docs are kept in sync with the README, and are an exact match. Is there something else we should do here?

We should move the user relevant content into the documentation, and reference the documentation as needed from the README files. This way, any updates can be made once in the docs.

**Comment by corey-derochie-amd** [comment]
> > We need to ensure that these docs are kept in sync with the README, and are an exact match. Is there something else we should do here?
> 
> We should move the user relevant content into the documentation, and reference the documentation as needed from the README files. This way, any updates can be made once in the docs.

I'd be okay including that change in this PR. Let me know if you'd like some cycles from us to help with that.

**Comment by randyh62** [comment]
> I'd be okay including that change in this PR. Let me know if you'd like some cycles from us to help with that.

I have added _Using NCCL Net Plugin_ as a topic in this PR. It was the only other README content I could find. Let me know if there is something more to do.

**Review by nileshnegi — APPROVED** [review]
LGTM

**Pull request merged**